### PR TITLE
Initial contribution

### PR DESCRIPTION
**Pull request by rzgry**
Initial contribution for vscode-microprofile. Starts up lsp4mp and supports contribution from external JARs. Also includes yaml support.

Not sure what I should do with version in package.json and the changelog.md file. Also did not touch the contributing.md so that will need to be updated.

Also wondering if we can remove the extension dependency on `vscjava.vscode-java-debug`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Quarkus Tools for Visual Studio Code Changelog
```
Comment (angelozerr) [inline CHANGELOG.md:1]: Rename to `MicroProfile Tools for Visual Studio Code Changelog`

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
```
Comment (angelozerr) [inline CHANGELOG.md:2]: Remove the whole content of the file

File: package.json [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+        "id": "microprofile-properties",
+        "aliases": [
+          "MicroProfile properties"
+        ],
+        "filenames": [
+          "microprofile-config.properties"
+        ],
+        "configuration": "./language-support/properties-support/language-configuration.json"
+      },
+      {
```
Comment (angelozerr) [inline package.json:67]: Remove this quarkus language declaration

File: package.json [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        }
+      }
+    },
+    "grammars": [
+      {
+        "language": "microprofile-properties",
+        "scopeName": "source.java-properties",
+        "path": "./language-support/properties-support/java-properties.tmLanguage.json"
+      },
+      {
```
Comment (angelozerr) [inline package.json:201]: For the moment, keep like this but we should remove this quarkus language declaration in an another PR

File: src/extension.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  context.subscriptions.push(registerConfigurationUpdateCommand());
+  context.subscriptions.push(registerOpenURICommand());
+}
+
+function connectToLS(context: ExtensionContext) {
+  return requirements.resolveRequirements().then(requirements => {
+    const clientOptions: LanguageClientOptions = {
+      documentSelector: [
+        { scheme: 'file', language: 'microprofile-properties' },
+        { scheme: 'file', language: 'quarkus-properties' },
```
Comment (angelozerr) [inline src/extension.ts:91]: Keep like this, but this quarkus document selector should be removed in an another PR.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,341 @@
+# Contribution Guide
+
+Contributions are extremely welcome, no matter how big or small.
+If you have any questions or suggestions we are happy to hear them.
+
+# Table of Contents
+1. [Project Structure](#project-structure)
+2. [Implementing language features for `application.properties`](#implementing-language-features-for-applicationproperties)
```
Comment (angelozerr) [inline CONTRIBUTING.md:8]: [Implementing language features for `microprofile-config.properties`](#implementing-language-features-for-microprofileconfigproperties)

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+    1. [Searching for properties in JARs not in the user's Quarkus project's classpath](#searching-for-properties-in-jars-not-in-the-users-quarkus-projects-classpath)
```
Comment (angelozerr) [inline CONTRIBUTING.md:9]: [Searching for properties in JARs not in the user's MicroProfile project's classpath](#searching-for-properties-in-jars-not-in-the-users-microprofile-projects-classpath)

File: package.json [hunk 8]
```diff
@@ -0,0 +1,253 @@
+{
+  "name": "vscode-microprofile",
+  "displayName": "MicroProfile Language Support",
+  "description": "MicroProfile Language Support for Visual Studio Code",
+  "version": "1.7.0",
```
Comment (angelozerr) [inline package.json:5]: "version": "0.1.0",

**Comment by angelozerr** [comment]
> Initial contribution for vscode-microprofile. Starts up lsp4mp and supports contribution from external JARs. Also includes yaml support.

I tested (without quarkus stuff and yaml stuff) and it works well.

> Not sure what I should do with version in package.json 

0.1.0 must be used (preview version)

> and the changelog.md file.

See my comments

> Also did not touch the contributing.md so that will need to be updated.

Keep like this for the moment, I create an issue for that https://github.com/redhat-developer/vscode-microprofile/issues/2

> Also wondering if we can remove the extension dependency on vscjava.vscode-java-debug

I think we need to keep those dependencies to debug Quarkus application. @xorye please correct me. Perhaps we can remove it for MicroProfile?

@rzgry please fix my comments and squash your commits in one commit, after that I'm good to merge it. The other update could be done in another PRs.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+This Visual Studio Code extension provides support for [MicroProfile](https://github.com/eclipse/microprofile) development via:
+
+ * a [MicroProfile language server](https://github.com/eclipse/lsp4mp/tree/master/microprofile.ls).
+ * a [MicroProfile jdt.ls extension](https://github.com/eclipse/lsp4mp/tree/master/microprofile.jdt).
+
+![](images/propertiesSupport.png)
+
+## MicroProfile `properties` Features
+
+In `application.properties` and `microprofile-config.properties` files, you will benefit with:
```
Comment (angelozerr) [inline README.md:14]: remove application.properties 'which is specfic to quarkus)

File: README.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+  * Hover support for MicroProfile
+  * Validation and Quick Fix support for MicroProfile
+  * Code Lens support for MicroProfile
+  * Code snippets
+
+## Requirements
+
+  * Java JDK (or JRE) 8 or more recent
+  * [Language Support for Java(TM) by Red Hat](https://marketplace.visualstudio.com/items?itemName=redhat.java)
+  * [Debugger for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-debug)
```
Comment (angelozerr) [inline README.md:37]: @xorye the debugger for Java is only required for Quarkus, no?

File: package.json [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+    "workspaceContains:**/src/main/resources/application.yaml",
+    "workspaceContains:**/src/main/resources/application.yml",
+    "onLanguage:microprofile-properties",
+    "onLanguage:quarkus-properties",
+    "onLanguage:java"
+  ],
+  "main": "./dist/extension",
+  "extensionDependencies": [
+    "redhat.java",
+    "vscjava.vscode-java-debug"
```
Comment (angelozerr) [inline package.json:44]: @xorye the debugger for Java is only required for Quarkus, no? Could we remove this dependency?

Comment (angelozerr) [reply package.json:44]: @rzgry you can remove this dependency

File: README.md (hunk 10 above)
Comment (angelozerr) [reply README.md:37]: @rzgry you can remove this dependency

**Comment by rzgry** [comment]
Removed dependency on vscode-java-debug and squashed my commits.

**Comment by angelozerr** [comment]
Thanks @rzgry !

File: README.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+  * Completion support for MicroProfile
+  * Hover support for MicroProfile
+  * Validation and Quick Fix support for MicroProfile
+  * Code Lens support for MicroProfile
+  * Code snippets
+
+## Requirements
+
+  * Java JDK (or JRE) 8 or more recent
```
Comment (fbricon) [inline README.md:35]: Java 11 is required

Comment (angelozerr) [reply README.md:35]: fixed https://github.com/redhat-developer/vscode-microprofile#requirements
